src/main/java/org/openbase/bco/bcozy/controller/powerterminal/chartcontroller/TilesFxChartController.java

```diff
@@ -45,9 +49,19 @@ public ScheduledFuture enableDataRefresh(long interval, ChartStateModel chartSta
 
     @Override
     public void updateChart(ChartStateModel chartStateModel) {
-        //todo: Update charts from within a thread with GlobalSched,,.submit() and call update from within Platform.runLater() see #92
-        this.view.getChartData().clear();
-        this.view.getChartData().setAll(UnitConverter.convert(chartStateModel.getUnit(), PowerTerminalDBService.getAverageConsumptionForDateRange(chartStateModel.getDateRange())));
+        GlobalScheduledExecutorService.submit(() -> {
```

Comment:
Please do not use the GlobalScheduledExecutorService for single task executions.
Please switch to GlobalCachedExecutorService (offers the same api)

Reply:
Fixed in e5b34de